### PR TITLE
chore: add datadog

### LINKED ISSUE #57: Add datadog
Similar to the usage in lndhub we should add datadog. 
It helps with operations and monitoring

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #57

### CODE REVIEW

**Comment by bumi** [comment]
\o/

**Review by kiwiidb — CHANGES_REQUESTED** [review]
Still needs sqltrace. cfr how we do it in lndhub: https://github.com/getAlby/lndhub.go/blob/d4e1217b8e63b9a8fc4c82095fec2aa7f0b5247c/db/db.go#L25

File: service/service.go [hunk 1]
```diff
@@ -82,16 +87,29 @@ func InitService(conf *Config) (svc *Service, err error) {
...
-		return nil, nil, nil, err
-	}
+	var sqlDb *sql.DB
+	if cfg.DatadogAgentUrl != "" {
+		sqltrace.Register("pgx", &stdlib.Driver{}, sqltrace.WithServiceName("oauth2server"))
+		sqlDb, err = sqltrace.Open("pgx", dsn)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		db, err = gormtrace.Open(postgres.New(postgres.Config{Conn: sqlDb}), &gorm.Config{})
```
Comment (kiwiidb) [inline service/service.go:97]: Why do you use a different Open argument here than on line 102? It can just be the same, no?

Comment (im-adithya) [reply service/service.go:97]: Then I wouldn't be using the sqlDb which is set for tracing right?

**Pull request merged**